pages/favorites.jsx

```diff
@@ -0,0 +1,80 @@
+import useLocalStorageState from "use-local-storage-state";
+import ActivityCard from "../components/ActivityCard";
+import styled from "styled-components";
+import useSWR from "swr";
+import { useEffect, useState } from "react";
+
+export default function Favorites() {
+  const [favoriteIds, setFavorites] = useLocalStorageState("favorites", {
+    defaultValue: [],
+  });
+
```

Comment:
```suggestion
const [localActivities, setLocalActivities] = useState([]);
```
Keep the activities list in local state and update it optimistically when an item is unfavorited, without relying on SWR to refetch it.